src/main/java/com/google/cloud/genomics/dataflow/utils/GCSFilename.java

```diff
@@ -0,0 +1,47 @@
+package com.google.cloud.genomics.dataflow.utils;
+
+import java.io.IOException;
+
+/**
+ * A holder for the (bucket, filename) pairs we get with Google Cloud Storage,
+ * plus helper code to translate between the (bucket,filename) and "gs://bucket/filename" formats.
+ */
+public class GCSFilename {
+  public static final String PREFIX = "gs://";
+  public final String bucket;
+  public final String filename;
+
+  /**
+   * Hold the (bucket,filename) pair.
+   * filename can include "/", bucket cannot.
+   */
+  public GCSFilename(String bucket, String filename) {
+    this.bucket = bucket;
+    this.filename = filename;
+  }
+
+  /**
+   * Parse a GCD URL and hold the result.
+   * gcsPathUrl is of the form gs://BUCKET/FILENAME. filename can include "/", bucket cannot.
+   * @throws IOException
+   */
+  public GCSFilename(String gcsPathUrl) throws IOException {
```

Comment:
I don't know of existing similar code, but it's possible it exists. I know
there's somewhat similar code in one of the GCS helper classes as part of
its operation. However I think there's value in having a class to do this
trivial kind of parsing without needing dependencies on anything else.

If anything, we may want to change the code in
GenomicsDatasetOptions:validateOptions to call into GCSFilename for
validation. What do you think?

On Thu, Apr 16, 2015 at 3:35 PM, Nicole Deflaux <notifications@github.com>
wrote:

> In src/main/java/com/google/cloud/genomics/dataflow/utils/GCSFilename.java
> <https://github.com/googlegenomics/dataflow-java/pull/55#discussion_r28556403>
> :
>
> > +
> > +  /**
> > +   * Hold the (bucket,filename) pair.
> > +   * filename can include "/", bucket cannot.
> > +   */
> > +  public GCSFilename(String bucket, String filename) {
> > +    this.bucket = bucket;
> > +    this.filename = filename;
> > +  }
> > +
> > +  /**
> > +   * Parse a GCD URL and hold the result.
> > +   * gcsPathUrl is of the form gs://BUCKET/FILENAME. filename can include "/", bucket cannot.
> > +   * @throws IOException
> > +   */
> > +  public GCSFilename(String gcsPathUrl) throws IOException {
>
> Is similar code available in a GCS java library we depend upon?
>
> We also have
> https://github.com/googlegenomics/dataflow-java/blob/master/src/main/java/com/google/cloud/genomics/dataflow/utils/GenomicsDatasetOptions.java#L72
>
> And it might be nice to refactor GCS path validation into a reusable
> method, if its not already available elsewhere. Note that for --output
> its better if the path does not already exist so that people do not
> accidentally over-write data, but we are not currently enforcing
> non-existence.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/googlegenomics/dataflow-java/pull/55/files#r28556403>.
>